src/Main.elm

```diff
@@ -1,10 +1,251 @@
+   the data, view would always display it accurately - on the opposite site, the Dict would grow outdated and would therefore need to be updated too, which
+   kind of defeats the purpose of creating it in the first place).
+-}
+{- | Use elm-live to launch the app. Install elm live and launch it with commands:
+
+   npm i -g elm-live
+   elm-live src/Main.elm --open -- --output main.js
+
+-}
+
+
+type alias Model =
+    { currentFolderId : Maybe Int
+    , folders : List AudienceFolder
+    , audiences : List Audience
+    , filter : Data.Audience.AudienceType
+    }
+
+
+type Msg
+    = Select Int
+    | GoUp (Maybe Int)
+    | Filter Data.Audience.AudienceType
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( { currentFolderId = Nothing
+      , folders =
+            --decode audienceFolder json into AudienceFolder List
+            case Decode.decodeString decodeDataAudienceFolder audienceFoldersJSON of
+                Ok res ->
+                    res
+
+                _ ->
+                    []
+      , audiences =
+            --decode audience json into Audience List
+            case Decode.decodeString decodeDataAudience audiencesJSON of
+                Ok res ->
+                    res
+
+                _ ->
+                    []
+      , filter = Data.Audience.Authored
+      }
+    , Cmd.none
+    )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        Select id ->
+            ( { model | currentFolderId = Just id }, Cmd.none )
+
+        GoUp id ->
+            ( { model | currentFolderId = id }, Cmd.none )
+
+        Filter type_ ->
+            ( { model | filter = type_, currentFolderId = Nothing }, Cmd.none )
+
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ -- display the buttons to filter by type_,
+          List.map
+            (\type_ ->
+                button
+                    [ onClick (Filter type_)
+                    , class "button-group-button"
+                    , if model.filter == type_ then
+                        style "opacity" "0.5"
+
+                      else
+                        style "opacity" "1"
+                    ]
+                    [ text (Debug.toString type_) ]
+            )
+            allTypes
+            |> div [ class "button-group" ]
+
+        --folders are handled here
+        , List.map
+            (\folder ->
+                let
+                    --is the current folder opened? (only one folder can be opened at a time)
+                    opened =
+                        case model.currentFolderId of
+                            Just id ->
+                                if folder.id == id then
+                                    True
+
+                                else
+                                    False
+
+                            Nothing ->
+                                False
```

Comment:
You can simply do
```suggestion
                        Just folder.id == model.currentFolderId
```

Reply:
Now that's a really nice way of writing it 👏. Lesson learned.